jobeet/src/jobeet/MyBundle/Manager/Manager.php

```diff
@@ -0,0 +1,210 @@
+
+use Doctrine\Entity;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityRepository;
+use jobeet\MyBundle\CacheDriver\AbstractCacheDriver;
+
+class Manager implements ManagerInterface
+{
+    /**
+     * @var AbstractCacheDriver
+     */
+    private $cacheDriver;
+
+    /**
+     * @var EntityManager
+     */
+    private $entityManager;
+
+    /**
+     * @var EntityRepository
+     */
+    private $repository;
+
+    /**
+     * @var int
+     */
+    private $flushInterval = 5;
+
+    /**
+     * @var String
+     */
+    private $entityClassPath;
+
+    /**
+     * @var String
+     */
+    private $entityClassName;
+
+    /**
+     * @param EntityManager $entityManager
+     * @param AbstractCacheDriver $cacheDriver
+     * @param String $classPath
+     * @param String $repositoryPath
+     */
+    public function __construct(EntityManager $entityManager, AbstractCacheDriver $cacheDriver, $classPath, $repositoryPath)
+    {
+        $this->entityManager = $entityManager;
+        $this->repository = $entityManager->getRepository($repositoryPath);
+        $this->entityClassPath = $classPath;
+        $this->cacheDriver = $cacheDriver;
+        $this->entityClassName = array_pop(explode("\\", $classPath));
+    }
+
+    /**
+     * @return int
+     */
+    public function getFlushInterval()
+    {
+        return $this->flushInterval;
+    }
+
+    /**
+     * @param int $flushInterval
+     */
+    public function setFlushInterval($flushInterval)
+    {
+        $this->flushInterval = $flushInterval;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function pushToDatabase(array $saveList )
+    {
+        $i = 0;
+        $this->entityManager->getConnection()->beginTransaction();
+        try {
+            foreach ($saveList as $entityToSave) {
+                $this->save($entityToSave);
+
+                if (++$i % $this->flushInterval == 0) {
+                    $this->entityManager->flush();
+                    $this->entityManager->getConnection()->commit();
+                }
+            }
+            $this->entityManager->flush();
+            $this->entityManager->getConnection()->commit();
+
+            $this->toCacheMultiple($saveList);
+        } catch (\Exception $e) {
+            $this->entityManager->getConnection()->rollBack();
+            throw $e;
+        }
+    }
+
+    private function save($entityToSave)
+    {
+        $this->entityManager->getConnection()->beginTransaction();
+
+        try {
+            $this->entityManager->persist($entityToSave);
+            $this->entityManager->getConnection()->commit();
+        } catch (\Exception $e) {
+            $this->entityManager->getConnection()->rollBack();
+            throw $e;
+        }
+    }
+
+    /**
+     * @param array $saveList
+     * @param int $ttl
+     * @throws \Exception
+     */
+    public function toCacheMultiple($saveList, $ttl = 600)
+    {
+        try {
+            foreach ($saveList as $entityToSave) {
```

Comment:
ezt a ciklust inkább a cacheDriver-nek kéne csinálnia, nem? 
Mártmint tudjon a cacheDriver collection-t is cachelni, hmm!?